apps/api/src/domain/models/digital-text/commands/create-digital-text.command-handler.ts

```diff
@@ -0,0 +1,146 @@
+import { IRepositoryProvider } from '../../../repositories/interfaces/repository-provider.interface';
+import { AggregateId } from '../../../types/AggregateId';
+import { InMemorySnapshot, ResourceType } from '../../../types/ResourceType';
+import buildInMemorySnapshot from '../../../utilities/buildInMemorySnapshot';
+import { BaseCommandHandler } from '../../shared/command-handlers/base-command-handler';
+import UuidNotGeneratedInternallyError from '../../shared/common-command-errors/UuidNotGeneratedInternallyError';
+import { BaseEvent } from '../../shared/events/base-event.entity';
+import { DigitalText } from '../digital-text.entity';
+import { CreateDigitalText } from './create-digital-text.command';
+import { DigitalTextCreated } from './digital-text-created.event';
+
+@CommandHandler(CreateDigitalText)
+export class CreateDigitalTextCommandHandler extends BaseCommandHandler<DigitalText> {
+    protected repositoryForCommandsTargetAggregate: IRepositoryForAggregate<DigitalText>;
+
+    constructor(
+        @Inject(REPOSITORY_PROVIDER_TOKEN)
+        protected readonly repositoryProvider: IRepositoryProvider,
+        @Inject('ID_MANAGER') protected readonly idManager: IIdManager
+    ) {
+        super(repositoryProvider, idManager);
+
+        this.repositoryForCommandsTargetAggregate =
+            this.repositoryProvider.forResource<DigitalText>(ResourceType.digitalText);
+    }
+
+    protected async createOrFetchWriteContext({
+        aggregateCompositeIdentifier: { id },
+        title,
+        languageCodeForTitle,
+    }: CreateDigitalText): Promise<ResultOrError<DigitalText>> {
+        const digitalTextDto: DTO<DigitalText> = {
+            id,
+            title: buildMultilingualTextWithSingleItem(title, languageCodeForTitle),
+            published: false,
+            type: ResourceType.digitalText,
+            eventHistory: [],
+        };
+
+        return getInstanceFactoryForResource<DigitalText>(ResourceType.digitalText)(digitalTextDto);
+    }
+
+    actOnInstance(digitalText: DigitalText): ResultOrError<DigitalText> {
+        return digitalText;
+    }
+
+    protected async fetchRequiredExternalState(): Promise<InMemorySnapshot> {
+        const searchResult = await this.repositoryProvider
+            .forResource<DigitalText>(ResourceType.digitalText)
+            .fetchMany();
+
+        const allDigitalTexts = searchResult.filter((digitalText): digitalText is DigitalText => {
+            if (isInternalError(digitalText)) {
+                throw digitalText;
+            }
+
+            return true;
+        });
+
+        // Deal with deprecation? Use DeluxeInMemoryStore?
+        return buildInMemorySnapshot({
+            resources: { digitalText: allDigitalTexts },
+        });
+    }
+
+    validateExternalState(state: InMemorySnapshot, digitalText: DigitalText): ValidationResult {
+        return digitalText.validateExternalState(state);
```

Comment:
BTW I am working on a pattern where this is not necessary. We can use the `@RefersTo` decorator metadata to perform this validation automatically. Note that if we were using a relational database and deferring this to the DB, this would be a referential integrity check. This should be just as automatic in our system. The only time to validate external state explicitly should be when you have a domain rule beyond existence \ ID uniqueness. For example, if two digital texts cannot share the same name, you would need to write the external state validation logic.